### ISSUE TITLE
Avoid null writes in deterministic component adjoints

### ISSUE THREAD

**Issue by daedalus5** [NVIDIA/warp#1945]
### Bug Description

A deterministic custom adjoint that accumulates into a component of an array adjoint can write through a null array descriptor when that input does not require gradients.

The update should be inactive when no backward destination exists. Instead, the component update can reach a direct slot store or atomic fallback with a null `adj_values.data`, causing a CUDA illegal-memory-access error and potentially poisoning the CUDA context.

#### Reproduction

```python
import warp as wp

wp.init()
wp.set_module_options({"deterministic": wp.DeterministicMode.RUN_TO_RUN})


@wp.func
def lookup(values: wp.array[wp.vec3], i: int) -> float:
    return values[i].x


@wp.func_grad(lookup)
def adj_lookup(values: wp.array[wp.vec3], i: int, adj_ret: float):
    wp.adjoint[values][i].x += adj_ret


@wp.kernel
def kernel(
    values: wp.array[wp.vec3],
    scale: wp.array[float],
    out: wp.array[float],
):
    i = wp.tid()
    out[i] = lookup(values, i) * scale[0]


device = "cuda:0"
n = 4
values = wp.ones(n, dtype=wp.vec3, requires_grad=False, device=device)
scale = wp.ones(1, dtype=float, requires_grad=True, device=device)
out = wp.zeros(n, dtype=float, requires_grad=True, device=device)

with wp.Tape() as tape:
    wp.launch(kernel, dim=n, inputs=[values, scale], outputs=[out], device=device)

tape.backward(grads={out: wp.ones_like(out)})
wp.synchronize_device(device)
```

#### Expected behavior

Backward execution completes. The nonexistent `values` adjoint update is skipped, while gradients for differentiable inputs such as `scale` are still computed.

#### Actual behavior

The custom component-adjoint update can dereference a null adjoint descriptor and trigger a CUDA illegal-memory-access error.

A regression test should cover a custom component adjoint where the source array has `requires_grad=False` but another input still requires gradients.

### System Information

- Warp: current `main` at `a05050a0836f79e44aceaf80dbea5c28be1c7450`
- Device: CUDA
- Deterministic mode: `RUN_TO_RUN` or `GPU_TO_GPU`
